assets/javascripts/job_templates.js

```diff
@@ -177,29 +181,60 @@ function buildMediumGroup(group, media) {
     });
 }
 
+function addArchSpacer(table, position, method) {
+	$(table).find('thead th.arch').eq(position)[method]('<th class="arch">&nbsp;</th>');
+	$(table).find('tbody tr').each(function(){
+		$(this).find('td.arch').eq(position)[method]('<td class="arch">&nbsp;</td>');
+	});
+}
+
 function alignCols() {
-    var maxwidth = 0;
-    $('td.name').each(function(index, test) {
-	if ($(this).width() > maxwidth) 
-	    maxwidth = $(this).outerWidth();
-    });
-    maxwidth = Math.ceil(maxwidth);
-    $('th.name').width(maxwidth);
-    maxarchs = 0;
-    $("table.mediagroup thead").each(function(index, th) {
-	var archs = $(this).find('th.arch').length;
-	if (archs > maxarchs) {
-	    maxarchs = archs;
-	}
+	// Set minimal width
+	$('th.name').width('0');
+	$('th.prio').width('0');
+	
+	// Find biggest minimal width
+	var namewidth = 450;
+	$('td.name').each(function(index, test) {
+		if ($(this).outerWidth() > namewidth) 
+			namewidth = $(this).outerWidth();
+	});
+	namewidth = Math.ceil(namewidth);
+
+    // Find header with all architectures
+    var ths_all = [];
+    $("table.mediagroup thead").each(function() {
+    	var archs = $(this).find('th.arch');
+    	if (archs.length > ths_all.length)
+    		ths_all = archs;
     });
-    var leftwidth = $('.jobtemplate-header').innerWidth() - $('th.name').outerWidth() - $('th.prio').outerWidth();
-
-    leftwidth = Math.floor(leftwidth / maxarchs) - 1;
-    $('th.arch').outerWidth(leftwidth);
-    leftwidth = Math.floor($('th.arch').width());
-    // round
-    $('th.arch').width(leftwidth);
-    return leftwidth;
-   
+
+	// Fill empty space
+	$("table.mediagroup").each(function(index, table) {
+		var ths = $(this).find('thead th.arch');
+		if (ths.length < ths_all.length) {
+			ths_all.each(function(i) {
+				// Used all ths, fill the rest
+				if (ths.length == i) {
+					for(var j = i; j < ths_all.length; j++) {
+						addArchSpacer(table, j-1, 'after');
+					}
+					return false;
+				} else if (this.innerHTML != ths.get(i).innerHTML) {
+					addArchSpacer(table, i, 'before');
+					ths = $(table).find('thead th.arch');
+				}
+			});
+		}
+	});
```

Comment:
I am filling empty architectures with <td>&nbsp;</td> so that columns are alligned properly for each architecture.

![screenshot from 2016-08-23 17-14-23](https://cloud.githubusercontent.com/assets/538604/17897622/1ba3e4e2-6955-11e6-9590-a23f28aa797c.png)